collator/src/mempool/impls/std_impl.rs

```diff
@@ -380,35 +290,57 @@ impl MempoolAdapter for MempoolAdapterStdImpl {
                 "Will start mempool with state update from mc block"
             );
 
-            if let Some((round, time)) = (config_guard.builder.get_genesis())
-                .filter(|(_, time)| *time > new_cx.consensus_info.genesis_millis)
-            {
+            if let Some(genesis) = (config_guard.builder.get_genesis()).filter(|genesis| {
+                genesis.start_round >= new_cx.consensus_info.genesis_round
+                    && genesis.time_millis > new_cx.consensus_info.genesis_millis
+            }) {
                 // Note: assume that global config is applied to mempool adapter
                 //   before collator is run in synchronous code, so this method is called later
 
+                // genesis does not have externals, so only strictly greater time and round
+                // will be saved into next block, so genesis can have values GEQ than in prev block
                 anyhow::ensure!(
-                    round >= new_cx.mc_processed_to_anchor_id && time >= new_cx.mc_block_chain_time,
-                    "new genesis round {} and time {} should be >= \
+                    genesis.start_round >= new_cx.mc_processed_to_anchor_id
+                        && genesis.time_millis >= new_cx.mc_block_chain_time,
+                    "new {genesis:?} should be >= \
                     master block processed_to_anchor_id {} and gen chain_time {}",
```

Comment:
this check makes change of genesis on the same start round secure.
note that next anchor round and time will be strictly greater than those in genesis.